neqo-http3/src/connection.rs

```diff
@@ -696,6 +696,13 @@ impl<E: Http3Events + Default, T: Http3Transaction, H: Http3Handler<E, T>>
         }
     }
 
+    fn state_closing(&self) -> bool {
+        match self.state {
```

Comment:
I wonder if we can start using `matches!(self.state, Http3State::Closing(_))` for these more.

Reply:
Seems like this is a feature that is in nightly (https://doc.rust-lang.org/nightly/std/macro.matches.html) unless we want to take it from an external crate. Either way, it would be so cool to be able to write

`!matches!(self.state, Http3State::Closing(_))`

rather than what's there now.

Do you want to include the https://docs.rs/matches/0.1.8/matches/macro.matches.html crate to do this?

Reply:
https://github.com/mozilla/neqo/blob/fa4b17afd26be8bf2c78cbc7247a31223b9cd2ca/neqo-common/src/lib.rs#L25 should do until we can get the one from `std`

Reply:
Thanks for pointing me to that. Sorry I didn't see it myself :-) Resolving this review -- see the new commit! Thanks again for the review!